lib/src/components/board/board.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:viam_sdk/src/resource/base.dart';
+
+import '../../gen/common/v1/common.pb.dart';
+import '../../robot/client.dart';
+
+abstract class Board extends Resource {
+  static const Subtype subtype = Subtype(ResourceNamespaceRDK, ResourceTypeComponent, "board");
+
+  @override
+  String name;
+
+  Board(this.name);
+
+  Future<void> setGPIO(String pin, bool high, {Map<String, dynamic>? extra});
+
+  // TODO: add rest of board methods
+
+  static ResourceName getResourceName(String name) {
+    return Board.subtype.getResourceName(name);
+  }
+
+  static Board fromRobot(RobotClient robot, String name) {
```

Comment:
What's the point of this method? Can users just call [`getResource`](https://github.com/viamrobotics/viam-flutter-sdk/blob/main/lib/src/robot/client.dart#L73-L75) directly and provide the type? I'm worried we'll forget to implement this in every resource. If we want it for convenience, is there a way to assert at compile time that every extension of `Resource` implemented `getResourceName` and `fromRobot`?

Reply:
Great point. There are probably multiple ways to handle that enforcement, the first that comes to mind is moving the definition of this method to `Resource` and have each class that inherits override it. 

Are you ok if I handle addressing this concern and changing this pattern in a follow up PR?

Reply:
> the first that comes to mind is moving the definition of this method to Resource and have each class that inherits override it.

that SGTM. And yeah feel free to address in follow-up PR.